### PR TITLE
add Update Method and fix project 

### PR DESCRIPTION
**Pull request by anovosiolovGD**
-erased "auth" from path
-turned off jwt token security while developing and testing features 
-added Update Method
-in pom.xml added validator dependency
-in .gitignore added local file

### CODE REVIEW

File: src/main/java/com/nvs/store/config/SecurityConfig.java [hunk 1]
```diff
@@ -17,22 +17,35 @@ public class SecurityConfig {
     private final JwtAuthFilter jwtAuthFilter;
     private final AuthenticationProvider authenticationProvider;
 
+//    @Bean
```
Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/config/SecurityConfig.java]: revert

Comment (ppanasencoGD) [reply src/main/java/com/nvs/store/config/SecurityConfig.java]: please remove security only in local development and testing - main functionality should have it

File: src/main/java/com/nvs/store/controllers/ProductController.java [hunk 2]
```diff
@@ -30,6 +28,10 @@ public ResponseEntity<Product> getProduct(@PathVariable Long id) {
     public ResponseEntity<Product> addProduct(@RequestBody Product product) {
         return productService.addProduct(product);
     }
+    @PutMapping("{id}")
+    public ResponseEntity<Product> updateProduct (@PathVariable Long id , @RequestBody Product product){
```
Comment (DrVa97) [inline src/main/java/com/nvs/store/controllers/ProductController.java]: Please use the formatting functionality from IDE to get rid of unnecessary spaces, and in general to format the source code files.

File: src/main/java/com/nvs/store/controllers/ProductController.java [hunk 3]
```diff
@@ -30,6 +28,10 @@ public ResponseEntity<Product> getProduct(@PathVariable Long id) {
     public ResponseEntity<Product> addProduct(@RequestBody Product product) {
         return productService.addProduct(product);
     }
+    @PutMapping("{id}")
```
Comment (ppanasencoGD) [inline src/main/java/com/nvs/store/controllers/ProductController.java]: please extract all ResponseEntities here from productService as as scope of conversion to response object should happen in controllers

**Review by ppanasencoGD — APPROVED** [review]

File: .gitignore [hunk 4]
```diff
@@ -19,6 +19,9 @@ target/
 *.iml
 *.ipr
 
+### Local files ###
+*.DS_Store
```
Comment (DrVa97) [inline .gitignore:23]: In order for this to be ignored, first you need to commit the change to the gitignore file. As you can see the .DS_Store file is not ignored.

**Review by DrVa97 — APPROVED** [review]

File: src/main/java/com/nvs/store/config/SecurityConfig.java (hunk 1 above)
Comment (anovosiolovGD) [reply src/main/java/com/nvs/store/config/SecurityConfig.java]: done

File: src/main/java/com/nvs/store/controllers/ProductController.java (hunk 3 above)
Comment (anovosiolovGD) [reply src/main/java/com/nvs/store/controllers/ProductController.java]: done

File: src/main/java/com/nvs/store/controllers/ProductController.java (hunk 2 above)
Comment (anovosiolovGD) [reply src/main/java/com/nvs/store/controllers/ProductController.java]: done

File: .gitignore (hunk 4 above)
Comment (anovosiolovGD) [reply .gitignore:23]: not done but I will find the way how to fix it

**Pull request merged**